gsy_framework/sim_results/kpi.py

```diff
@@ -306,60 +309,56 @@ def area_performance_indices(self, area_dict: Dict, core_stats: Dict) -> Dict:
-            "total_energy_demanded_wh": self.state[area_dict["uuid"]].total_demand,
+            "name": area_dict["name"],
+            "self_sufficiency": self_sufficiency,
+            "self_consumption": self_consumption,
+            "total_energy_demanded_wh": total_energy_demanded_wh,
             "demanded_buffer_wh": self.state[area_dict["uuid"]].demanded_buffer_wh,
             "self_consumption_buffer_wh": self.state[area_dict["uuid"]].self_consumption_buffer_wh,
             "total_energy_produced_wh": self.state[area_dict["uuid"]].total_energy_produced_wh,
             "total_self_consumption_wh": self.state[area_dict["uuid"]].total_self_consumption_wh,
         }
-        if self.savings_state.get(area_dict["uuid"]) is not None:
-            kpi_parm_dict.update(self.savings_state[area_dict["uuid"]].to_dict())
+
+        kpi_parm_dict.update(self.savings_state[area_dict["uuid"]].to_dict())
+
         return kpi_parm_dict
 
-    def _kpi_ratio_to_percentage(self, area_name: str):
-        area_kpis = self.performance_indices[area_name]
+    def _get_ui_formatted_results(self, area_uuid: str) -> Dict:
+        area_kpis = copy(self.performance_indices[area_uuid])
         if area_kpis["self_sufficiency"] is not None:
-            self_sufficiency_percentage = area_kpis["self_sufficiency"] * 100
+            area_kpis["self_sufficiency"] = area_kpis["self_sufficiency"] * 100
         else:
-            self_sufficiency_percentage = 0.0
+            area_kpis["self_sufficiency"] = 0.0
         if area_kpis["self_consumption"] is not None:
-            self_consumption_percentage = area_kpis["self_consumption"] * 100
+            area_kpis["self_consumption"] = area_kpis["self_consumption"] * 100
         else:
-            self_consumption_percentage = 0.0
-
-        return {"self_sufficiency": self_sufficiency_percentage,
-                "self_consumption": self_consumption_percentage,
-                "total_energy_demanded_wh": area_kpis["total_energy_demanded_wh"],
-                "demanded_buffer_wh": area_kpis["demanded_buffer_wh"],
-                "self_consumption_buffer_wh": area_kpis["self_consumption_buffer_wh"],
-                "total_energy_produced_wh": area_kpis["total_energy_produced_wh"],
-                "total_self_consumption_wh": area_kpis["total_self_consumption_wh"],
-                "base_case_cost": area_kpis.get("base_case_cost"),
-                "utility_bill": area_kpis.get("utility_bill"),
-                "fit_revenue": area_kpis.get("fit_revenue"),
-                "gsy_e_cost": area_kpis.get("gsy_e_cost"),
-                "saving_absolute": area_kpis.get("saving_absolute"),
-                "saving_percentage": area_kpis.get("saving_percentage")
-                }
-
-    # pylint: disable=arguments-renamed
+            area_kpis["self_consumption"] = 0.0
+
+        # to be consistent with the old version, where no name is present in the kpi structure
+        area_kpis.pop("name", None)
```

Comment:
Couldn't this be done by the frontend? We could send them everything and then the FE decides what to show/hide.

Reply:
Probably the entire logic of `_get_ui_formatted_results()` could/should be performed by the frontend, if it's only used to format data for the UI.

Reply:
I would be up for it, yes 👍